wazo_calld/config.py

```diff
@@ -1,15 +1,23 @@
 # Copyright 2015-2024 The Wazo Authors  (see the AUTHORS file)
 # SPDX-License-Identifier: GPL-3.0-or-later
 
+from __future__ import annotations
 
 import argparse
+from typing import TYPE_CHECKING
 
 from xivo.chain_map import ChainMap
 from xivo.config_helper import parse_config_file, read_config_file_hierarchy
 from xivo.xivo_logging import get_log_level_by_name
 
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+    from typing import Any
+
+    from wazo_calld.types import CalldConfigDict
```

Comment:
@DrPyser Out of curiosity, do you know if it's a good practice to make every typing import in the TYPE_CHECKING?
From what I understand, it should only be used on import conflict:
- collections: built-in library, so not useful
- typing: already imported by TYPE_CHECKING, so not useful
- wazo_calld.type: already in the same module than this code, so not useful

Reply:
My thinking was:
1. indicate that those imports are *only* for typing
2. reduce imports to the strict minimum when running production code

Reply:
Yep totally agree with you
But it's not the strategy used for other typed modules.
ex: https://github.com/python/cpython/blob/a07cf4ce25205d836a6b30cb629eda139d2a518e/Doc/tools/extensions/patchlevel.py#L6

Reply:
Well after search, it seems useless to add `from typing` in this block
But other imports seem to make sense
For the self types module ... I'm note sure, but I can live with it :smile:

Reply:
Otherwise I won't object in this case, I just wouldn't care to lobby for this pattern everywhere.

Reply:
Thanks for your feedback, I'll remove the conditional import unless strictly necessary then.